amplify-tools/amplify-tools-gradle-plugin/src/main/groovy/com/amplifyframework/tools/gradle/plugin/amplifytools.groovy

```diff
@@ -0,0 +1,246 @@
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.apache.tools.ant.taskdefs.condition.Os
+import groovy.json.JsonSlurper
+import groovy.json.JsonOutput
+
+class amplifytools implements Plugin<Project> {
+    void apply(Project project) {
+        def doesNodeExist = true
+        def doesGradleConfigExist
+
+        // profile name can be changed in amplify-gradle-config
+        def profile = 'default'
+        def accessKeyId = null
+        def secretAccessKey = null
+        def region = null
+        def envName = 'amplify'
+        def syncEnabled = 'true'
+
+        project.task('verifyNode') {
+            try {
+                project.exec {
+                    commandLine 'node', '-v'
```

Comment:
You actually run `npx`, not `node`, later. So, it'd be better to check if `npx` is available.

When I used this plugin, `npx` _wasn't_ available.

I do have `node` installed, but that's only _part_ of the requirement.